src/main/kotlin/com/rooplor/classcraftbackend/controllers/UserController.kt

```diff
@@ -0,0 +1,90 @@
+package com.rooplor.classcraftbackend.controllers
+
+import com.rooplor.classcraftbackend.dtos.Response
+import com.rooplor.classcraftbackend.dtos.UserRequest
+import com.rooplor.classcraftbackend.entities.User
+import com.rooplor.classcraftbackend.services.UserService
+import io.swagger.v3.oas.annotations.Operation
+import org.modelmapper.ModelMapper
+import org.springframework.http.ResponseEntity
+import org.springframework.web.bind.annotation.DeleteMapping
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.PathVariable
+import org.springframework.web.bind.annotation.PostMapping
+import org.springframework.web.bind.annotation.PutMapping
+import org.springframework.web.bind.annotation.RequestBody
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.RestController
+
+@RestController
+@RequestMapping("/api/user")
+class UserController(
+    private val userService: UserService,
+    private val modelMapper: ModelMapper,
+) {
+    @Operation(summary = "Get all users")
+    @GetMapping
+    fun getAllUsers(): ResponseEntity<Response<List<User>>> {
+        try {
+            val user = userService.findAllUser()
+            return ResponseEntity.ok(Response(success = true, result = user, error = null))
+        } catch (e: Exception) {
+            return ResponseEntity.badRequest().body(Response(success = false, result = null, error = e.message))
+        }
+    }
+
+    @Operation(summary = "Get user by id")
+    @GetMapping("/{id}")
+    fun getUserById(
+        @PathVariable id: String,
+    ): ResponseEntity<Response<User>> {
+        try {
+            val user = userService.findUserById(id)
+            return ResponseEntity.ok(Response(success = true, result = user, error = null))
+        } catch (e: Exception) {
+            return ResponseEntity.badRequest().body(Response(success = false, result = null, error = e.message))
+        }
+    }
+
+    @Operation(summary = "Insert a new user")
+    @PostMapping("/create")
```

Comment:
For best practice, I think we should not use `create` as endpoint name. endpoint should be noun not verb. My advice is use POST /api/user.